xonsh/aliases.py

```diff
@@ -75,7 +76,7 @@ def source_foreign(args, stdin=None):
     ns = parser.parse_args(args)
     if ns.prevcmd is not None:
         pass  # don't change prevcmd if given explicitly
-    elif os.path.isfile(ns.files_or_code[0]):
+    elif os.path.isfile(ns.files_or_code[0]) and args[0] != 'cmd':
```

Comment:
I think that this is because 'cmd' should be passed in as the sourcer, rather than as the first args element.